### PR TITLE
#263 dialog based verification

### PR DESCRIPTION
**Pull request by elKei24**
Includes #264.

Goal is to decrease steps necessary for an employee to verify cards of customers and improve style of the verification UI.

As verification is more a linear process than the other tabs, I think it is valid to not have a regular page at all, but instead just show the scanner page, and then a waiting message and the result as a popup. I put the information into a separate popup that opens before the scanner, but when the intro slides are done, we can reuse the mechanism there to just show that information as often until the user clicks "do not show again", so that it then only needs a single tap to verify cards.

### CODE REVIEW

**Comment by Schedulaar** [comment]
According to your aim, it would be nice if the dialog "So prüfen Sie die Echtheit einer Ehrenamtskarte" would have a button "Nicht mehr anzeigen" to dismiss this dialog forever and instantly go ahead to the camera.
We could have a "?" button on the app bar on the camera screen that shows the dialog again.

File: frontend/lib/verification/dialogs/verification_info_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: Text("So prüfen Sie die Echtheit einer Ehrenamtskarte"),
+      content: SingleChildScrollView(
+        child: ListBody(children: [
+          Text("Scannen Sie den QR-Code, der auf der „Ausweisen“-Seite Ihres "
+              "Gegenübers angezeigt wird. "
+              "Daraufhin wird durch eine Server-Anfrage geprüft, ob die "
+              "gescannte Ehrenamtskarte gültig ist. "
+              "Sie benötigen eine Internetverbindung."),
```
Comment (Schedulaar) [inline frontend/lib/verification/dialogs/verification_info_dialog.dart:14]:
```suggestion
          Text("Scannen Sie den QR-Code, der auf der „Ausweisen“-Seite Ihres "
              "Gegenübers angezeigt wird. "
              "Daraufhin wird durch eine Server-Anfrage geprüft, ob die "
              "gescannte Ehrenamtskarte gültig ist. "
              "Dazu wird eine Internetverbindung benötigt."),
```

File: frontend/lib/verification/verification_workflow.dart [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            Text("Wir überprüfen den gescannten Code für Sie."),
+            SizedBox(height: 12),
+            CircularProgressIndicator()
+        ]),
+        actions: [
+          TextButton(
+              onPressed: () => Navigator.pop(context),
+              child: Text("Abbrechen"))
+        ],
+      );
```
Comment (Schedulaar) [inline frontend/lib/verification/verification_workflow.dart:111]: Too much text for something that is usually displayed <1 second.
Maybe only show the ProgressIndicator and maybe the cancel button in a custom Dialog with opaque background. (or similar :) )

**Review by Schedulaar — CHANGES_REQUESTED** [review]

**Comment by elKei24** [comment]
> According to your aim, it would be nice if the dialog "So prüfen Sie die Echtheit einer Ehrenamtskarte" would have a button "Nicht mehr anzeigen" to dismiss this dialog forever and instantly go ahead to the camera.
> We could have a "?" button on the app bar on the camera screen that shows the dialog again.

Yes, I am waiting for the intro slides before I add that, because we would both add the shared preferences etc otherwise

File: frontend/lib/verification/dialogs/verification_result_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          icon: Icons.verified_user,
+          iconColor: Colors.green,
+          child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                Table(
+                  children: [
+                    toTableRow("Name: ", cardDetails.fullName),
+                    toTableRow("Ablaufdatum: ", expirationDateString),
+                    toTableRow("Landkreis: ", cardDetails.regionId.toString()),
```
[the remark this answers is not in the document]
Comment (elKei24) [reply frontend/lib/verification/dialogs/verification_result_dialog.dart:69]: Ah s**t, that must have got lost during the merge yesterday

File: frontend/lib/verification/dialogs/verification_result_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+              children: [
+                Text(reason),
+                if (errorCode != null) ...[
+                  SizedBox(height: 8),
+                  Text(
+                    "Fehlercode: $errorCode",
+                    textScaleFactor: 0.5,
+                    style: TextStyle(color: theme.disabledColor),
+                  ),
+            ]])));
```
[the remark this answers is not in the document]
Comment (elKei24) [reply frontend/lib/verification/dialogs/verification_result_dialog.dart]: Done

File: frontend/lib/verification/dialogs/verification_result_dialog.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (elKei24) [reply frontend/lib/verification/dialogs/verification_result_dialog.dart]: Fixed it

File: frontend/lib/verification/verification_workflow.dart (hunk 2 above)
Comment (elKei24) [reply frontend/lib/verification/verification_workflow.dart]: True. I simplified it.

**Review by Schedulaar — APPROVED** [review]

File: frontend/lib/verification/dialogs/verification_result_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                Table(
+                  children: [
+                    toTableRow("Name: ", cardDetails.fullName),
+                    toTableRow("Ablaufdatum: ", expirationDateString),
+                    toTableRow("Landkreis: ", cardDetails.regionId.toString()),
+                    toTableRow("Typ: ", cardDetails.cardType == CardType.gold
+                      ? "Gold" : "Standard"),
```
[the remark this answers is not in the document]
Comment (elKei24) [reply frontend/lib/verification/dialogs/verification_result_dialog.dart]: Fixed it

**Comment by elKei24** [comment]
> Yes, I am waiting for the intro slides before I add that, because we would both add the shared preferences etc otherwise

@Schedulaar As the intro slides were merged in the meantime, I just added the option to hide the info text.

**Comment by Schedulaar** [comment]
@elKei24 nice! If you want you can add a "?"-button in the title bar when the scanner is active to get the help back :)

**Review by Schedulaar — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
Ill test https://github.com/ehrenamtskarte/ehrenamtskarte/pull/307 instead of this one

**Pull request merged**